pkg/resource/resource_share/hooks.go

```diff
@@ -134,36 +134,36 @@ func (rm *resourceManager) syncPermissions(
 	desiredPermissions := desired.ko.Spec.PermissionARNs
 	latestPermissions := latest.ko.Spec.PermissionARNs
 
-	toAdd, toDelete := comparePermissionArns(desiredPermissions, latestPermissions)
+	toAdd, toDelete := compareStringSlices(desiredPermissions, latestPermissions)
 
-	if len(toDelete) > 0 {
-		rlog.Debug("disassociating permissions from ResourceShare resource", "permissionArns", toDelete)
-		for _, permission := range toDelete {
-			_, err = rm.sdkapi.DisassociateResourceSharePermissionWithContext(
+	if len(toAdd) > 0 {
```

Comment:
It would nice to have a comment/PR description explaining why we need to add before deleting.